### PR TITLE
Introduce interpolation mode to vectro and quaternion keys

### PR DESCRIPTION
**Pull request by kimkulling**

### CODE REVIEW

**Comment by tellypresence** [comment]
Refactoring USD code should be in separate PR since it doesn't directly relate to animation interpolation

File: code/AssetLib/USD/USDLoader.cpp [hunk 1]
```diff
@@ -1,43 +1,46 @@
...
+A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+@author: Richard Steffen, 2014
```
Comment (tellypresence) [inline code/AssetLib/USD/USDLoader.cpp:40]: Remove incorrect author

File: code/AssetLib/USD/USDLoader.cpp [hunk 2]
```diff
@@ -87,8 +90,7 @@ USDImporter::USDImporter() :
 
 // ------------------------------------------------------------------------------------------------
 
-bool USDImporter::CanRead(const std::string &pFile, IOSystem *pIOHandler, bool checkSig) const {
-    UNUSED(checkSig);
```
[the remark this answers is not in the document]
Comment (kimkulling) [reply code/AssetLib/USD/USDLoader.cpp:91]: This is right, I need to remove the variable declaration from the signature as well.

File: code/AssetLib/USD/USDLoaderImplTinyusdz.cpp [hunk 3]
```diff
@@ -81,9 +81,7 @@ using namespace std;
 void USDImporterImplTinyusdz::InternReadFile(
         const std::string &pFile,
         aiScene *pScene,
-        IOSystem *pIOHandler) {
-    UNUSED(pIOHandler);
-    UNUSED(TAG); // Ignore unused variable when -Werror enabled
```
[the remark this answers is not in the document]
Comment (kimkulling) [reply code/AssetLib/USD/USDLoaderImplTinyusdz.cpp:86]: Same here.

**Pull request merged**